File: Python/Kode/PetShop.py
class PetShop:
    def __init__(self):
        self._ids = []
        self._names = []
        self._categories = []
        self._prices = []
    
    # Getter untuk mendapatkan semua data dalam format yang sesuai
    def get_products(self):
        products = []
        for i in range(len(self._ids)):
            product = type('Product', (), {
                'id': self._ids[i],
                'nama': self._names[i],
                'kategori': self._categories[i],
                'harga': self._prices[i]
            })
            products.append(product)
        return products
    
    # Getter untuk masing-masing list
    def get_ids(self):
        return self._ids
    
    def get_names(self):
        return self._names
    
    def get_categories(self):
        return self._categories
    
    def get_prices(self):
        return self._prices
    
    # Setter untuk mengubah data berdasarkan index
    def set_name(self, index, value):
        if 0 <= index < len(self._names):
            self._names[index] = value
    
    def set_category(self, index, value):
        if 0 <= index < len(self._categories):
            self._categories[index] = value
    
    def set_price(self, index, value):
        if 0 <= index < len(self._prices):
            self._prices[index] = value
    
    # Method untuk menambah produk
    def add_product(self, id, nama, kategori, harga):
        self._ids.append(id)
        self._names.append(nama)
        self._categories.append(kategori)
        self._prices.append(harga)
    
    # Method untuk menghapus produk
    def remove_product(self, id):
        for i in range(len(self._ids)):
            if self._ids[i] == id:
                self._ids.pop(i)
                self._names.pop(i)
                self._categories.pop(i)
                self._prices.pop(i)
                return True
        return False
    
    # Method untuk mengubah data produk
    def update_product(self, id, nama, kategori, harga):
        for i in range(len(self._ids)):
            if self._ids[i] == id:
                self._names[i] = nama
                self._categories[i] = kategori
                self._prices[i] = harga
                return True
        return False

File: PHP/Kode/petshop.php
<?php
// Deklarasi class PetShop
class PetShop {
    // Deklarasi property untuk menyimpan data produk
    private $ids = array();         // Menyimpan ID produk
    private $names = array();       // Menyimpan nama produk
    private $categories = array();  // Menyimpan kategori produk
    private $prices = array();      // Menyimpan harga produk
    private $photos = array();      // Menyimpan foto produk

    // Method untuk menambah produk baru
    public function addProduct($name, $category, $price, $photo) {
        // Generate ID baru berdasarkan jumlah produk yang ada
        $id = count($this->ids) + 1;

        // Simpan data produk ke dalam array yang sesuai
        $this->ids[] = $id;
        $this->names[] = $name;
        $this->categories[] = $category;
        $this->prices[] = $price;
        $this->photos[] = $photo;
    }

    // Method untuk mendapatkan semua ID produk
    public function getIds() {
        return $this->ids;
    }

    // Method untuk mendapatkan semua nama produk
    public function getNames() {
        return $this->names;
    }

    // Method untuk mendapatkan semua kategori produk
    public function getCategories() {
        return $this->categories;
    }

    // Method untuk mendapatkan semua harga produk
    public function getPrices() {
        return $this->prices;
    }

    // Method untuk mendapatkan semua foto produk
    public function getPhotos() {
        return $this->photos;
    }

    // Method untuk menghapus produk berdasarkan ID
    public function deleteProduct($id) {
        // Cari index produk berdasarkan ID
        $index = array_search($id, $this->ids);

        // Jika produk ditemukan, hapus data pada semua array
        if ($index !== false) {
            unset($this->ids[$index]);
            unset($this->names[$index]);
            unset($this->categories[$index]);
            unset($this->prices[$index]);
            unset($this->photos[$index]);

            // Reindex array agar urutannya rapih kembali
            $this->ids = array_values($this->ids);
            $this->names = array_values($this->names);
            $this->categories = array_values($this->categories);
            $this->prices = array_values($this->prices);
            $this->photos = array_values($this->photos);
        }
    }

    // Method untuk mengubah data produk berdasarkan ID
    public function editProduct($id, $name, $category, $price) {
        // Cari index produk berdasarkan ID
        $index = array_search($id, $this->ids);

        // Jika produk ditemukan, update data produk
        if ($index !== false) {
            $this->names[$index] = $name;
            $this->categories[$index] = $category;
            $this->prices[$index] = $price;
        }
    }
}
?>
